collector.go

```diff
@@ -210,60 +227,140 @@ func getMapReduceIndexReducedTotal(stats *stats) float64 {
 	return value
 }
 
-func getDatabaseDocuments(dbStats *dbStats) float64 {
+func getDatabaseDocuments(dbStats *dbStats) []metricInfo {
+	var mi []metricInfo
+
 	value, _ := jp.GetFloat(dbStats.databaseStats, "CountOfDocuments")
-	return value
+	labels := generateDatabaseLabels(dbStats, nil)
+	mi = appendMetricInfo(mi, value, labels)
+
+	return mi
 }
 
-func getDatabaseIndexes(dbStats *dbStats) float64 {
+func getDatabaseIndexes(dbStats *dbStats) []metricInfo {
+	var mi []metricInfo
 	value, _ := jp.GetFloat(dbStats.databaseStats, "CountOfIndexes")
-	return value
+	labels := generateDatabaseLabels(dbStats, nil)
+	mi = appendMetricInfo(mi, value, labels)
+
+	return mi
 }
 
-func getDatabaseStaleIndexes(dbStats *dbStats) float64 {
+func getDatabaseTasks(dbStats *dbStats) []metricInfo {
+	var mi []metricInfo
+
+	// Required for compatibility with versions 5 and 6
+	onGoingTasksKey := "OngoingTasks"
+	_, _, _, err := jp.Get(dbStats.tasks, onGoingTasksKey)
+	if err != nil {
+		onGoingTasksKey = "OngoingTasksList"
+	}
+
+	taskAggregate := make(map[string]float64)
+	jp.ArrayEach(dbStats.tasks, func(value []byte, dataType jp.ValueType, offset int, err error) {
+		taskType, _ := jp.GetString(value, "TaskType")
+		connectionStatus, _ := jp.GetString(value, "TaskConnectionStatus")
+
+		// unique key based on task type and connection status for aggregation
+		key := taskType + "_" + connectionStatus
+
+		taskAggregate[key] += 1
+
+		labels := generateDatabaseLabels(dbStats, map[string]string{
+			"type":              taskType,
+			"connection_status": connectionStatus,
+		})
+
+		mi = appendMetricInfo(mi, taskAggregate[key], labels)
```

Comment:
Here, each key will be represented with multiple entries in `mi`. This works ok because later on these entries are applied in order to prometheus metric, so only the last value is actually reported on the metric. However it may be confusing for readers, who are not aware of this mechanism. How about a little bit different approach here:

```go
func getDatabaseTasks(dbStats *dbStats) []metricInfo {
	var mi []metricInfo

	// Required for compatibility with versions 5 and 6
	onGoingTasksKey := "OngoingTasks"
	_, _, _, err := jp.Get(dbStats.tasks, onGoingTasksKey)
	if err != nil {
		onGoingTasksKey = "OngoingTasksList"
	}

	type key struct {
		taskType, connectionStatus string
	}

	taskAggregate := make(map[key]float64)

	jp.ArrayEach(dbStats.tasks, func(value []byte, dataType jp.ValueType, offset int, err error) {
		taskType, _ := jp.GetString(value, "TaskType")
		connectionStatus, _ := jp.GetString(value, "TaskConnectionStatus")
		taskAggregate[key{taskType, connectionStatus}] += 1
	}, onGoingTasksKey)

	for k, v := range taskAggregate {
		labels := generateDatabaseLabels(dbStats, map[string]string{
			"type":              k.taskType,
			"connection_status": k.connectionStatus,
		})

		mi = appendMetricInfo(mi, v, labels)
	}

	return mi
}
```

Reply:
Thank you for the suggestion, looks cleaner indeed. https://github.com/marcinbudny/ravendb_exporter/pull/6/commits/0700516b2a1e601decda840b9960aa7fd6776ad1